api/app/sfms/daily_fwi_processor.py

```diff
@@ -20,48 +20,49 @@
 MultiDatasetContext = Callable[[List[str]], Iterator[List["WPSDataset"]]]
 
 
-class BUIDateRangeProcessor:
+class DailyFWIProcessor:
     """
-    Class for calculating/generating forecasted DMC/DC/BUI rasters for a date range
+    Class for calculating/generating forecasted daily FWI rasters for a date range
     """
 
     def __init__(self, start_datetime: datetime, days: int, addresser: RasterKeyAddresser):
         self.start_datetime = start_datetime
         self.days = days
         self.addresser = addresser
 
-    async def process_bui(self, s3_client: S3Client, input_dataset_context: MultiDatasetContext, new_dmc_dc_context: MultiDatasetContext):
+    async def process(self, s3_client: S3Client, input_dataset_context: MultiDatasetContext, new_dmc_dc_context: MultiDatasetContext):
         set_s3_gdal_config()
 
         for day in range(self.days):
             datetime_to_calculate_utc, previous_fwi_datetime, prediction_hour = self._get_calculate_dates(day)
             logger.info(f"Calculating DMC/DC/BUI for {datetime_to_calculate_utc.isoformat()}")
 
             # Get and check existence of weather s3 keys
-            temp_key, rh_key, _, precip_key = self.addresser.get_weather_data_keys(self.start_datetime, datetime_to_calculate_utc, prediction_hour)
+            temp_key, rh_key, wind_speed_key, precip_key = self.addresser.get_weather_data_keys(self.start_datetime, datetime_to_calculate_utc, prediction_hour)
             weather_keys_exist = await s3_client.all_objects_exist(temp_key, rh_key, precip_key)
             if not weather_keys_exist:
                 logging.warning(f"No weather keys found for {model_run_for_hour(self.start_datetime.hour):02} model run")
                 break
 
             # get and check existence of fwi s3 keys
-            dc_key, dmc_key = self._get_previous_fwi_keys(day, previous_fwi_datetime)
+            dc_key, dmc_key, ffmc_key = self._get_previous_fwi_keys(day, previous_fwi_datetime)
             fwi_keys_exist = await s3_client.all_objects_exist(dc_key, dmc_key)
             if not fwi_keys_exist:
                 logging.warning(f"No previous DMC/DC keys found for {previous_fwi_datetime.date().isoformat()}")
                 break
 
-            temp_key, rh_key, precip_key = self.addresser.gdal_prefix_keys(temp_key, rh_key, precip_key)
+            temp_key, rh_key, wind_speed_key, precip_key = self.addresser.gdal_prefix_keys(temp_key, rh_key, wind_speed_key, precip_key)
             dc_key, dmc_key = self.addresser.gdal_prefix_keys(dc_key, dmc_key)
```

Comment:
I think ffmc might need to be included here

Reply:
Great catch!